Remove GenericCallback.h

achristensen07 opened the pull request:
#### b5fdd5c2d78718547db4a78999f57ee6c6773df1
<pre>
Remove GenericCallback.h
<a href="https://bugs.webkit.org/show_bug.cgi?id=252179">https://bugs.webkit.org/show_bug.cgi?id=252179</a>
rdar://105406323

Reviewed by Chris Dumez.

Also remove some more dead code, including StatisticsRequest.cpp and
decidePolicyForSpeechRecognitionPermissionRequest which is not called.

* Source/WebKit/Shared/ImageOptions.h:
* Source/WebKit/UIProcess/API/APIAttachment.h:
* Source/WebKit/UIProcess/API/APIIconLoadingClient.h:
* Source/WebKit/UIProcess/API/APIInspectorExtension.cpp:
* Source/WebKit/UIProcess/API/APIUIClient.h:
(API::UIClient::decidePolicyForSpeechRecognitionPermissionRequest): Deleted.
* Source/WebKit/UIProcess/API/C/WKPage.cpp:
(WKPageSetPageUIClient):
* Source/WebKit/UIProcess/API/C/WKSpeechRecognitionPermissionCallback.cpp:
(WKSpeechRecognitionPermissionCallbackComplete):
* Source/WebKit/UIProcess/API/Cocoa/WKBrowsingContextController.mm:
* Source/WebKit/UIProcess/API/Cocoa/WKErrorInternal.h:
* Source/WebKit/UIProcess/Automation/WebAutomationSession.cpp:
* Source/WebKit/UIProcess/Cocoa/IconLoadingDelegate.mm:
* Source/WebKit/UIProcess/Cocoa/UIDelegate.h:
* Source/WebKit/UIProcess/Cocoa/UIDelegate.mm:
(WebKit::UIDelegate::UIClient::decidePolicyForSpeechRecognitionPermissionRequest): Deleted.
* Source/WebKit/UIProcess/DrawingAreaProxy.h:
* Source/WebKit/UIProcess/GenericCallback.h: Removed.
* Source/WebKit/UIProcess/Inspector/WebInspectorUIExtensionControllerProxy.cpp:
* Source/WebKit/UIProcess/Notifications/WebNotificationManagerProxy.cpp:
* Source/WebKit/UIProcess/StatisticsRequest.cpp: Removed.
* Source/WebKit/UIProcess/StatisticsRequest.h: Removed.
* Source/WebKit/UIProcess/WebBackForwardList.h:
* Source/WebKit/UIProcess/WebContextClient.cpp:
* Source/WebKit/UIProcess/WebFrameProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebProcessPool.cpp:
* Source/WebKit/UIProcess/WebProcessPool.h:
* Source/WebKit/UIProcess/mac/WKFullScreenWindowController.h:
* Source/WebKit/UIProcess/mac/WKTextFinderClient.mm:
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteLayerTreeContext.mm:
* Tools/WebKitTestRunner/TestController.cpp:
(WTR::TestController::createWebViewWithOptions):
(WTR::decidePolicyForSpeechRecognitionPermissionRequest): Deleted.
(WTR::TestController::completeSpeechRecognitionPermissionCheck): Deleted.
* Tools/WebKitTestRunner/TestController.h:

Canonical link: <a href="https://commits.webkit.org/260291@main">https://commits.webkit.org/260291@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/386824ec7ea44d4c0b4b9bd546ec2bcb4849b36a

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/107780 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/16836 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/40674 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/116924 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/116302 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/111670 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/18246 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/8151 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/99959 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/113537 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/13777 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/96977 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/41479 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/95684 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/28618 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/83322 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/9796 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/29966 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/10486 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/6859 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/15950 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/49550 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/12050 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/3862 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

cdumez reviewed (approved):
LGTM as long as EWS is happy.

achristensen07 committed:
Remove GenericCallback.h
https://bugs.webkit.org/show_bug.cgi?id=252179
rdar://105406323

Reviewed by Chris Dumez.

Also remove some more dead code, including StatisticsRequest.cpp and
decidePolicyForSpeechRecognitionPermissionRequest which is not called.

* Source/WebKit/Shared/ImageOptions.h:
* Source/WebKit/UIProcess/API/APIAttachment.h:
* Source/WebKit/UIProcess/API/APIIconLoadingClient.h:
* Source/WebKit/UIProcess/API/APIInspectorExtension.cpp:
* Source/WebKit/UIProcess/API/APIUIClient.h:
(API::UIClient::decidePolicyForSpeechRecognitionPermissionRequest): Deleted.
* Source/WebKit/UIProcess/API/C/WKPage.cpp:
(WKPageSetPageUIClient):
* Source/WebKit/UIProcess/API/C/WKSpeechRecognitionPermissionCallback.cpp:
(WKSpeechRecognitionPermissionCallbackComplete):
* Source/WebKit/UIProcess/API/Cocoa/WKBrowsingContextController.mm:
* Source/WebKit/UIProcess/API/Cocoa/WKErrorInternal.h:
* Source/WebKit/UIProcess/Automation/WebAutomationSession.cpp:
* Source/WebKit/UIProcess/Cocoa/IconLoadingDelegate.mm:
* Source/WebKit/UIProcess/Cocoa/UIDelegate.h:
* Source/WebKit/UIProcess/Cocoa/UIDelegate.mm:
(WebKit::UIDelegate::UIClient::decidePolicyForSpeechRecognitionPermissionRequest): Deleted.
* Source/WebKit/UIProcess/DrawingAreaProxy.h:
* Source/WebKit/UIProcess/GenericCallback.h: Removed.
* Source/WebKit/UIProcess/Inspector/WebInspectorUIExtensionControllerProxy.cpp:
* Source/WebKit/UIProcess/Notifications/WebNotificationManagerProxy.cpp:
* Source/WebKit/UIProcess/StatisticsRequest.cpp: Removed.
* Source/WebKit/UIProcess/StatisticsRequest.h: Removed.
* Source/WebKit/UIProcess/WebBackForwardList.h:
* Source/WebKit/UIProcess/WebContextClient.cpp:
* Source/WebKit/UIProcess/WebFrameProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebProcessPool.cpp:
* Source/WebKit/UIProcess/WebProcessPool.h:
* Source/WebKit/UIProcess/mac/WKFullScreenWindowController.h:
* Source/WebKit/UIProcess/mac/WKTextFinderClient.mm:
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteLayerTreeContext.mm:
* Tools/WebKitTestRunner/TestController.cpp:
(WTR::TestController::createWebViewWithOptions):
(WTR::decidePolicyForSpeechRecognitionPermissionRequest): Deleted.
(WTR::TestController::completeSpeechRecognitionPermissionCheck): Deleted.
* Tools/WebKitTestRunner/TestController.h:

Canonical link: https://commits.webkit.org/260291@main
